### ISSUE TITLE
What is the state of null-safety support in Flame at the moment?

### ISSUE THREAD

**Issue by lig** [flame-engine/flame#545]
I couldn't find any info on the null-safety support state in Flame.

Is it worth attempting to work on that?

**Comment by spydon** [comment]
I think the plan is to have it before we release the final v1.0.0 version.

**Comment by lig** [comment]
@spydon is the work started already? I would like to give it a try myself. AFAIU, first all the dependencies need to have the NS support anyway.

**Comment by spydon** [comment]
@lig is is not started, so please go ahead an give it a try, it would be very appreciated!
I think the `OrderedSet` needs to be updated first, if they haven't done that already.

**Comment by lig** [comment]
@spydon yep, `OrderedSet` it is and, I've checked lately, there is no support for NS there yet.

**Comment by spydon** [comment]
If you want to you could put up a PR in for that one too, shouldn't be that much code, it is maintained by @luanpotter.

**Comment by lig** [comment]
@spydon that's exactly what I'm planning to do. Go to every dependency and then up to Flame (sic).

**Comment by lig** [comment]
@spydon I think it could be useful to create a branch for `null-safety` ready code as it shouldn't be considered stable yet according to https://dart.dev/null-safety/migration-guide#step5-publish

It would be handy to target PRs addressing null-safety to such a branch, IMO.

BTW, https://github.com/luanpotter/ordered_set/pull/5

**Comment by spydon** [comment]
@lig good idea
https://github.com/flame-engine/flame/tree/null-safety

**Comment by lig** [comment]
https://github.com/flame-engine/flame/pull/549
